corehq/apps/sms/models.py

```diff
@@ -1435,3 +1436,210 @@ def initiate_workflow(cls, domain, phone_numbers, app_id=None,
+        """
+        result = self.get_extra_fields()
+        for k, v in kwargs.iteritems():
+            if k not in self.get_available_extra_fields():
+                raise Exception("Field %s is not an available extra field for %s"
+                    % (k, self.__class__.__name__))
+            result[k] = v
+
+        self.extra_fields = json.dumps(result)
+
+    def set_shared_domains(self, domains):
+        if self.id is None:
+            raise Exception("Please call .save() on the backend before "
+                "calling set_shared_domains()")
+        with transaction.atomic():
+            self.mobilebackendinvitation_set.all().delete()
+            self.mobilebackendinvitation_set = [
+                MobileBackendInvitation(
+                    domain=domain,
+                    accepted=True,
+                ) for domain in domains
+            ]
+        # TODO: Remove the below line once the two-way sync with
+        # couch is no longer necessary.
+        self.save()
+
+    def soft_delete(self):
+        with transaction.atomic():
+            self.deleted = True
+            self.mobilebackendinvitation_set.all().delete()
+            for mapping in self.sqlmobilebackendmapping_set.all():
+                # TODO: Can do a bulk delete once the two-way sync
+                # with couch is no longer necessary
+                mapping.delete()
+            self.save()
+
+    def _migration_sync_to_couch(self, couch_obj):
+        couch_obj.domain = self.domain
+        couch_obj.name = self.name
+        couch_obj.display_name = self.display_name
+        couch_obj.authorized_domains = [
+            i.domain for i in self.mobilebackendinvitation_set.all()
+        ]
+        couch_obj.is_global = self.is_global
+        couch_obj.description = self.description
+        couch_obj.supported_countries = json.loads(self.supported_countries)
+        couch_obj.reply_to_phone_number = self.reply_to_phone_number
+        couch_obj.backend_type = self.backend_type
+        couch_obj.reply_to_phone_number = self.reply_to_phone_number
+        for k, v in self.get_extra_fields().iteritems():
+            setattr(couch_obj, k, v)
+
+        load_balancing_numbers = json.loads(self.load_balancing_numbers)
+        if load_balancing_numbers:
+            couch_obj.x_phone_numbers = load_balancing_numbers
+
+        if self.deleted:
+            if not couch_obj.base_doc.endswith('-Deleted'):
+                couch_obj.base_doc += '-Deleted'
+
+        couch_obj.save(sync_to_sql=False)
+
+
+class SQLSMSBackend(SQLMobileBackend):
+    class Meta:
+        proxy = True
+
+
+class SQLMobileBackendMapping(SyncSQLToCouchMixin, models.Model):
+    """
+    A SQLMobileBackendMapping instance is used to map SMS or IVR traffic
+    to a given backend based on phone prefix.
+    """
+    class Meta:
+        db_table = 'messaging_mobilebackendmapping'
+
+    couch_id = models.CharField(max_length=126, null=True, db_index=True)
```

Comment:
this field gets populated directly from couch so i don't think it's necessary to make it a `UUIDField`, but i did update the `inbound_api_key` field to be a `UUIDField` which seems like a better fit for it